components/script/dom/htmlinputelement.rs

```diff
@@ -81,7 +86,7 @@ pub struct HTMLInputElement {
     // https://html.spec.whatwg.org/multipage/#concept-input-value-dirty-flag
     value_dirty: Cell<bool>,
 
-    // TODO: selected files for file input
+    filelist: DOMRefCell<Option<JS<FileList>>>,
```

Comment:
Oh, I see, it's an option because not all text inputs are file inputs, my bad.

Yeah, just use mutnullableheap here. It's not a primitive interface, but it's basically `RefCell<Option<T>>` where `T` is `JS<U>`. It's using DOMRefCell above because TextInput isn't a JS thing.